database/validators.py

```diff
@@ -16,3 +16,14 @@ def student_id_validator(sid):
     """
     if len(str(sid)) != 7:
         raise ValidationError("Student ID must be 7 digits long.")
+
+def year_validator(year):
+    """
+    This validator will ensure that the provided year is within a reasonable range
+    """
+    if year>(timezone.now().year+100) or year<(timezone.now().year-100):
+        raise ValidationError("Year out of range. Please select a more appropriate range.")
+
+def max_integer_validator(num):
+    if len(str(num))>7:
```

Comment:
The highest (theoretical value) for a course number is 999. Perhaps you can just check if the passed number is larger than 999 or smaller than 0?

Reply:
I wasn't sure if we should keep it more open if they change how to catalog courses. I am fine changing this

Reply:
That's fair